mautrix_telegram/user.py

```diff
@@ -831,13 +831,28 @@ async def _post_sync_dialog(
         try:
             if last_read:
                 await puppet.intent.mark_read(last_read.mx_room, last_read.mxid)
-            if was_created or not self.config["bridge.tag_only_on_create"]:
-                await self._mute_room(puppet, portal, mute_until)
-                await self._tag_room(puppet, portal, self.config["bridge.pinned_tag"], pinned)
-                await self._tag_room(puppet, portal, self.config["bridge.archive_tag"], archived)
+
+            # Tagging is handled by the portal, so we only need to handle changing the mute
+            # state/tags in the update case here.
```

Comment:
Getting dialogs with limit 10 isn't going to handle tagging if the portal isn't in the most recent 10 dialogs 🤔

Reply:
If it's not a backfill, then this should always work since it should bubble it up to the top of the chat list.

Reply:
The problem is that the same methods are used for backfill too